src/ape/api/projects.py

```diff
@@ -0,0 +1,259 @@
+        Most often, this is either a version number or a branch name.
+        """
+
+    @property
+    def _target_manifest_cache_file(self) -> Path:
+        version_id = self.version_id
+        if isinstance(
+            version_util.parse(version_id), version_util.Version
+        ) and not version_id.startswith("v"):
+            version_id = f"v{version_id}"
+
+        name = self.name
+        return self._data_folder / "packages" / name / version_id / f"{name}.json"
+
+    @abstractmethod
+    def extract_manifest(self) -> PackageManifest:
+        """
+        Create a :class:`~ape.api.projects.ProjectAPI` implementation,
+        presumably by downloading and compiling the dependency.
+
+        Implementations may use ``self.project_manager`` to call method
+        :meth:`~ape.managers.project.ProjectManager.get_project`
+        to dynamically get the correct :class:`~ape.api.projects.ProjectAPI`.
+        based on the project's structure.
+
+        Returns:
+            :class:`~ape.api.projects.ProjectAPI`
+        """
+
+    @property
+    def cached_manifest(self) -> Optional[PackageManifest]:
+        """
+        The manifest from the ``.ape/packages/<dependency-name>/<version-id>``
+        if it exists and is valid.
+        """
+
+        return _load_manifest_from_file(self._target_manifest_cache_file)
+
+    def _extract_local_manifest(self, project_path: Path):
+        contracts_folder = project_path / self.contracts_folder
+        project = self.project_manager.get_project(
+            project_path,
+            contracts_folder=contracts_folder,
+            name=self.name,
+            version=self.version,
+        )
+        sources = [
+            s
+            for s in get_all_files_in_directory(project.contracts_folder)
+            if s.name.lower() not in ("package.json", "package-lock.json")
+        ]
+        project_manifest = project.create_manifest(file_paths=sources)
+
+        # Cache the manifest for future use outside of this tempdir.
+        self._target_manifest_cache_file.parent.mkdir(exist_ok=True, parents=True)
+        self._target_manifest_cache_file.write_text(json.dumps(project_manifest.dict()))
+
+        return project_manifest
+
+
+def _load_manifest_from_file(file_path: Path) -> Optional[PackageManifest]:
+    if not file_path.exists():
+        return None
+
+    def handle_corrupt_manifest():
+        logger.warning(f"Existing manifest file '{file_path}' corrupted (re-building).")
+        file_path.unlink()
+
+    try:
+        manifest_dict = json.loads(file_path.read_text())
+        if not isinstance(manifest_dict, dict) or "manifest" not in manifest_dict:
+            handle_corrupt_manifest()
+            return None
+
+    except json.JSONDecodeError:
+        handle_corrupt_manifest()
+        return None
+
+    try:
+        return PackageManifest(**manifest_dict)
+    except ValidationError:
+        handle_corrupt_manifest()
+        return None
```

Comment:
```suggestion
    try:
        manifest_dict = json.loads(file_path.read_text())
        assert isinstance(manifest_dict, dict) and "manifest" in manifest_dict
        return PackageManifest(**manifest_dict)
    except (ValidationError, json.JSONDecodeError, AssertionError):
        logger.warning(f"Existing manifest file '{file_path}' corrupted (re-building).")
        file_path.unlink()
        return None
```

Reply:
`assert` statements is only for debug / testing, they really shouldn't be used in python production code.
If you set `__debug__ = False` (like via `python -O`), the `assert` statements get stripped out of there.

Reply:
I will try to refactor this to a single except block nonetheless!